src/PointwiseFunctions/Hydro/EquationsOfState/Enthalpy.cpp

```diff
@@ -0,0 +1,451 @@
+  // Preprocessing to put coefficients in better basis c_i z^i  = c_i' z^i/i!
+  std::vector<double> taylor_series_coefficients(
+      polynomial_coefficients.size());
+  for (size_t i = 0; i < integral_poly_coeffs.size(); i++) {
+    taylor_series_coefficients[i] =
+        polynomial_coefficients[i] * static_cast<double>(factorial(i));
+  }
+  // i indexes elements of the derivative, r sum over lower degree terms
+  for (size_t i = 0; i < taylor_series_coefficients.size(); i++) {
+    for (size_t r = 0; r <= i; r++) {
+      integral_poly_coeffs[r] +=
+          ((i - r) % 2 == 0 ? 1 : -1) * taylor_series_coefficients[i];
+    }
+    // restore the default normalization for the coefficients
+  }
+
+  for (size_t r = 0; r < integral_poly_coeffs.size(); r++) {
+    integral_poly_coeffs[r] /= static_cast<double>(factorial(r));
+  }
+
+  // note sum starts from 1
+  for (size_t j = 0; j < sin_coefficients.size(); j++) {
+    // contribution from the sine terms
+    double k = trig_scale;
+    integral_sin_coeffs[j] +=
+        1.0 / (square(j + 1) * square(k) + 1.0) * sin_coefficients[j];
+    integral_cos_coeffs[j] -= (static_cast<double>(j + 1) * k) /
+                              (square(j + 1) * square(k) + 1.0) *
+                              sin_coefficients[j];
+    // contribution from the cosine terms
+    integral_cos_coeffs[j] +=
+        1.0 / (square(j + 1) * square(k) + 1.0) * cos_coefficients[j];
+    integral_sin_coeffs[j] += (static_cast<double>(j + 1) * k) /
+                              (square(j + 1) * square(k) + 1.0) *
+                              cos_coefficients[j];
+  }
+  exponential_integral_coefficients.has_exponential_prefactor = true;
+  exponential_integral_coefficients.exponential_external_constant = 0.0;
+  exponential_integral_coefficients.polynomial_coefficients =
+      integral_poly_coeffs;
+  exponential_integral_coefficients.sin_coefficients = integral_sin_coeffs;
+  exponential_integral_coefficients.cos_coefficients = integral_cos_coeffs;
+  double new_constant = initial_condition.second -
+                        evaluate_coefficients(exponential_integral_coefficients,
+                                              initial_condition.first);
+  exponential_integral_coefficients.exponential_external_constant =
+      new_constant;
+  return exponential_integral_coefficients;
+}
+template <typename LowDensityEoS>
+typename Enthalpy<LowDensityEoS>::Coefficients
+Enthalpy<LowDensityEoS>::Coefficients::compute_derivative() {
+  // Slow/Bad?
+  std::vector<double> derivative_poly_coeffs(polynomial_coefficients.size());
+  std::vector<double> derivative_sin_coeffs(sin_coefficients.size());
+  std::vector<double> derivative_cos_coeffs(cos_coefficients.size());
+  Enthalpy::Coefficients derivative_coefficients = *this;
+  // d/dz e^z \sum_i a_i f(z) = e^z \sum_i a_i f_i(z)  + e^z \sum_i a_i f_i'(z)
+  if (has_exponential_prefactor) {
+    ERROR("You shouldn't have to do this!");
```

Comment:
Ah Sorry this was a debugging thing (i.e. it wasn't needed "right now", and there's no code that currently calls it).  It could very well be useful in the future, though I don't know exactly how.

Reply:
okay, sounds like maybe it's fine to remove then? If you had an idea of how it colud be used and think it would be reasonable to use then I'd say it's fine to keep. Does that sound okay?